modules/internal/ChapelArray.chpl

```diff
@@ -2625,10 +2629,20 @@ module ChapelArray {
     /* The number of dimensions in the array */
     proc rank param return this.domain.rank;
 
-    /* return the array's indices as its domain */
-    pragma "return not owned"
-    proc indices
+    proc indices where arrayIndicesAlwaysLocal == false {
+      compilerWarning("In a forthcoming release, calling '.indices' on an array will return/yield its indices locally rather than returning a copy of the array's domain as it does today.  See https://github.com/chapel-lang/chapel/issues/17883 for details.  To preserve the current behavior, use '.domain' or a copy thereof.  To update to the proposed new behavior today, re-compile with '-sarrayIndicesAlwaysLocal=true'.");
```

Comment:
Can you reflow this to fit in 80 columns?

I know that we don't do it very strictly, but I hope that we can. This warning is just way too long :)

Reply:
Just to clarify:  Are you asking that the source text fit within 80 columns (which I agree with) or the output text (which I'd also agree with except that it's always challenging given that you don't know how many columns the input filename will take up... we really need an error message pretty-printer to handle such things).

Reply:
I was asking about the source code.

Reply:
I did this, but then moved the warning to a deprecation warning which, AFAIK, doesn't support splitting the warning across multiple lines, :(